noir_stdlib/src/hash/poseidon.nr

```diff
@@ -0,0 +1,100 @@
+use crate::array;
+use crate::pow_32;
+mod bn254; // Instantiations of Poseidon for prime field of the same order as BN254
+
+// General Poseidon permutation on elements of type Field
+fn permute<N,O,P>(rf: u8, // Number of full rounds; should be even
+                  rp: u8, // Number of partial rounds
+                  alpha: Field, // S-box power, depending on the underlying field
+                  ark: [Field; O], // Additive round keys
+                  mds_matrix: [Field; P], // MDS Matrix in row-major order
+                  state: [Field; N] // State as an array of length t = c + r
+) -> [Field; N]
+{
+    let t = array::len(state);
+
+    // Input checks
+    constrain t as u8 * (rf + rp) == array::len(ark) as u8;
+    constrain t * t == array::len(mds_matrix);
+    constrain alpha > 0;
+    
+    let mut count = 0;
+
+    let mut out_perm = state;
+
+    for r in 0..(array::len(ark)/array::len(state)) // i.e. r in 0..rf + rp
+    {
```

Comment:
Looks like a different curly brace style than we use in the rest of the stdlib. Can we change it to the same line `for ... {`?